mutt_attach.c

```diff
@@ -1149,3 +1148,29 @@ int mutt_print_attachment(FILE *fp, struct Body *a)
     return 0;
   }
 }
+
+/* 2 small functions to handle temporary files */
+
+/* this function just adds a filename to the tail of TempAttachmentsList */
+void mutt_add_temp_attachment(char *filename)
+{
+  struct ListNode *n;
+
+  n = mutt_list_insert_tail(&TempAttachmentsList, NULL);
+  n->data=mutt_str_strdup(filename);
+}
+
+/* this function iterates through TempAttachmentsList and deletes all files */
+void mutt_unlink_temp_attachments(void)
+{
+  struct ListNode *np, *tmp;
+
+  STAILQ_FOREACH_SAFE(np, &TempAttachmentsList, entries, tmp)
+  {
+    mutt_file_chmod_add(np->data, S_IWUSR);
+    mutt_file_unlink(np->data);
+    STAILQ_REMOVE(&TempAttachmentsList, np, ListNode, entries);
+    FREE(&np->data);
+    FREE(&np);
+  } 
+}
```

Comment:
This is just a suggestion...
I think I'd have used a plain `STAILQ_FOREACH` to do the work,
then called `mutt_list_free()` to clean up the list, afterwards.